File: experimental/golang/hello_test.go
package hello

import "testing"

func TestExtractWorld(t *testing.T) {
	got, ok := ExtractWorld(Hello)
	if !ok {
		t.Fatalf("expected to find world")
	}
	if got != "world" {
		t.Fatalf("got %q, want %q", got, "world")
	}
}

func TestExtractWorldMissing(t *testing.T) {
	if _, ok := ExtractWorld("no match"); ok {
		t.Fatalf("expected missing world")
	}
}

File: devops/experimental/lua/tests/hello_test.lua
package.path = table.concat({
  package.path,
  string.format("%s/../src/?.lua", debug.getinfo(1, "S").source:match("^@(.+)/"))
}, ";")

local hello = require("hello")

assert(hello.extract_world(hello.HELLO) == "world", "should find world")
assert(hello.extract_world("no match") == nil, "should return nil when missing")

print("[ok] lua tests passed")
